persistence/src/main/java/com/epam/esm/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.entity.Pagination;
+import com.epam.esm.entity.User;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * @author Ivan Velichko
+ * @date 30.10.2021 13:45
+ */
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    public UserDaoImpl(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    @Override
+    public List<User> findAll(Pagination pagination) {
+        CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+        CriteriaQuery<User> criteriaQuery = criteriaBuilder.createQuery(User.class);
+        Root<User> user = criteriaQuery.from(User.class);
+        criteriaQuery.select(user);
+        return entityManager.createQuery(criteriaQuery).getResultList();
+    }
+
+    @Override
+    public User create(User user) {
+        entityManager.persist(user);
+        return user;
```

Comment:
In my opinion, it would be better to return the object from the database by id. Your object can be updated during a save operation (listeners, triggers, etc.)

Reply:
fixed